docs/learn/learn-xcvm.md

```diff
@@ -8,28 +8,108 @@ slug: ../learn-xcvm
+register-based state machine. The state is tracked in domain-specific registers that can hold
+information regarding the temporary state of a particular message. Most of the XCM format comprises
+these registers and the instructions used to compose XCVM programs.
+
+The XCVM is an ultra-high-level non-Turing-complete computer whose instructions are designed to be
+roughly at the same level as transactions in terms of definition. Messages are one or more XCM
+instructions executed in order by the XCVM. An XCM is executed until it either runs to the end or
+hits an error, at which point it finishes up and halts.
+
+The primary implementation of the XCVM is the
+[`xcm-executor`](https://github.com/paritytech/polkadot/tree/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor).
+It follows the XCVM specification provided by Parity. It's engineered to be extendable, providing
+maximum customizability when configuring XCM. Because the `xcm-executor` is just an implementation
+of XCVM, it's entirely possible to create another implementation if desired.
+
+## XCMs are XCVM Programs
+
+A cross consensus message (XCM) is just a program that runs on the `XCVM`: in other words, one or
+more XCM instructions that are executed by the `xcm-executor`. To learn more about the XCVM and the
+XCM format, see the latest
+[blog post on XCM v3](https://medium.com/polkadot-network/xcm-part-iii-execution-and-error-management-ceb8155dd166)
 by Dr. Gavin Wood.
 
-XCM instructions might change a register, they might change the state of the consensus system or
-both.
+XCM instructions might change a register, the state of the consensus system, or both. Depending on
+the program's goal, whether it is to teleport assets from one chain to another or call a smart
+contract on another chain, XCMs usually require changes to the registers before any changes to the
+consensus system can be made.
+
+## XCM Executor & Configuration
+
+The XCM Executor's implementation centers around a core piece: the XCM configuration. Each instance
+of the Executor must have a valid configuration, which specifies a multitude of options on how a
+chain may treat incoming messages via
+[Barriers](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L52),
+calculate weight for a message via the
+[Weigher](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L55),
+how much weight to purchase via the
+[Trader](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L58),
+[configure fees](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L89),
+how to
+[convert origins](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L40),
+and more.
```

Comment:
We should have a section in the wiki for all the possible configurations. I don't think we have one right now. A short introduction here is the way to go though

Reply:
Agreed - in the works after the high level stuff is done :D